xbmc/windows/GUIMediaWindow.cpp

```diff
@@ -900,7 +900,10 @@ bool CGUIMediaWindow::OnClick(int iItem, const std::string &player)
     else if (!CProfilesManager::GetInstance().GetCurrentProfile().canWriteSources() && !g_passwordManager.IsProfileLockUnlocked())
       return false;
 
-    return OnAddMediaSource();
+    if (OnAddMediaSource())
+      Refresh();
+
+    return true;
```

Comment:
It is supposed to return true if the action was taken. The result of the action is irrelevant regarding the return value. I am 100% sure that this is done incorrectly at other location because I recently fixed one of those already.
When you return false, the parent window or some other client may try to take the action again.